singularity/quit.sh

```diff
@@ -7,14 +7,16 @@ set -eo pipefail
 cd /sw/apps/dpdash
 npm stop
 
-# TODO: make this do a graceful exit
+# kill PM2
+./node_modules/pm2/bin/pm2 kill
 
-pkill -f 'pm2'
+# kill supervisord (thus mongo, celery, erlang, rabbitmq)
+echo "Quitting supervisord..."
+pidfile="/data/dpdash/supervisord/supervisord.pid"
+pid=$(<$pidfile)
+kill -SIGQUIT $pid
 
-# TODO: send SIGQUIT to supervisord
-
-pkill -f 'supervisord'
-pkill -f 'rabbitmq'
-pkill -f 'erlang'
-pkill -f 'celery'
-pkill -f 'mongo'
+while kill -s 0 $pid ; do : ; sleep 0.25 ; done
```

Comment:
So after every killing is done, are you waiting 0.25 seconds?

Reply:
I got this line from [this StackOverflow answer](https://askubuntu.com/a/1103089). In this line I'm just checking every 0.25 seconds if the process exists. That loop will go indefinitely until the process is successfully shutdown.

This is because `supervisord` takes a while to quit. It's indefinite how long that will take, so I just loop to check if it still exists until it's fully quit.

That way, when someone runs `run.sh`, they won't get an error that the port is already used.